controllers/authContorller.js

```diff
@@ -63,13 +67,88 @@ exports.signUp = async (req, res) => {
   }
 }
 
-exports.signIn = (req, res) => {
+export const signIn = (req, res) => {
   // everything is okay - send token!
   const token = getToken(req.user.id)
 
   res.send({ token })
 }
 
-exports.getCurrentUser = (req, res) => {
+export const getCurrentUser = (req, res) => {
   res.send(_.pick(req.user, ['id', 'username', 'email', 'created_at']))
 }
+
+export const requestReset = async (req, res) => {
+  const { email } = req.body
+  // make sure that user exists
+  const user = await db('users')
+    .select('*')
+    .where('email', email || null)
+    .first()
+
+  if (!user)
+    return res.status(404).send(`No account found for ${email || 'that'} email`)
+
+  // set a reset token and expiry on that user
+  const randomBytesPromiseified = promisify(randomBytes)
+
+  await db('users')
+    .where('email', email)
+    .update({
+      reset_token: (await randomBytesPromiseified(20)).toString('hex'),
+      reset_token_expiry: Date.now() + 3600000 // 1 hour from now
+    })
+
+  const resetURL = `http://${req.headers.host}/api/account/reset/${
+    user.reset_token
+  }`
+
+  // email them with reset token
+  await mailTransport.sendMail({
+    from: 'noreply@spotify-dev.com',
+    to: user.email,
+    subject: 'Your Password Reset Token',
+    html: mailTemplate(`Your Password Reset Token is here:
+      \n\n
+      <a href="${resetURL}">Click Here to Reset</a>`)
+  })
+
+  res.send({ message: 'Success!' })
+}
+
+export const resetPassword = async (req, res) => {
+  const { password, confirmPassword } = req.body
+  const { token } = req.params
+
+  // check if the passwords match and its not empty
+  if (password == null || confirmPassword == null)
+    return res.status(422).send('Passwords cannot be blank')
+
+  if (password !== confirmPassword)
+    return res.status(422).send("Passwords don't match")
+
+  // check if the token is valid and has not expired
+  const user = await db('users')
+    .where('reset_token', token)
+    .andWhere('reset_token_expiry', '>', Date.now() - 3600000)
```

Comment:
move magic numbers to config or .env file (360000), this is second duplicate

Reply:
Also move DB operations out of controller

Reply:
Done both!